ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx

```diff
@@ -1,5 +1,117 @@
+import { Button, Col, Notification, Row } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+
+import ControllerLink from "app/base/components/ControllerLink";
+import Definition from "app/base/components/Definition";
 import TitledSection from "app/base/components/TitledSection";
+import fabricSelectors from "app/store/fabric/selectors";
+import type { Fabric } from "app/store/fabric/types";
+import type { RootState } from "app/store/root/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+import { getFullVLANName } from "app/store/vlan/utils";
+import subnetsURLs from "app/subnets/urls";
+import { isId } from "app/utils";
+
+type Props = {
+  id: VLAN[VLANMeta.PK] | null;
+};
+
+// Note this is not the same as the getDHCPStatus VLAN util as it uses slightly
+// different language and renders a link for the relayed VLAN.
+const getDHCPStatus = (vlan: VLAN, vlans: VLAN[], fabrics: Fabric[]) => {
+  if (vlan.dhcp_on) {
+    return "Enabled";
+  }
+  if (vlan.external_dhcp) {
+    return `External (${vlan.external_dhcp})`;
+  }
+  if (isId(vlan.relay_vlan)) {
+    return (
+      <span>
+        Relayed via{" "}
+        <Link to={subnetsURLs.vlan.index({ id: vlan.relay_vlan })}>
+          {getFullVLANName(vlan.relay_vlan, vlans, fabrics)}
+        </Link>
+      </span>
+    );
+  }
+  return "Disabled";
+};
+
+const DHCPStatus = ({ id }: Props): JSX.Element | null => {
```

Comment:
Should this also fetch the vlans, fabrics and subnets? I wonder if that would be better to do in this component as someone could easily remove it from the parent component and this would break.

Reply:
It could, but I'm of two minds about including all the `fetch()` actions even if a parent component already does that. Taken to its logical extreme we should be running `fetch()` or `get()` functions for every component that retrieves something from state (e.g. in table cell components), which to me seems cumbersome. And in some cases where we're passing an `id` prop down 4 or 5 levels, are we meant to include `fetch()` at every step? But yeah, it does mean the component can be moved anywhere and "just work". I dunno, what do you think?

Reply:
Yeah I think it's a balancing act. `fetch()` is cheap because it's cached. In this case I was particularly thinking about the fact you added `dispatch(fabricActions.fetch());` in the parent component but its not data is not used by that component so I can imagine someone in the future thinking it's no longer needed and remove it.

Reply:
That's fair. I'll add them in.